utils/aws/route53/__init__.py

```diff
@@ -0,0 +1,318 @@
+        :type account: Account
+        :raises DuplicateException: if an account by that name already exists
+        """
+        if self.accounts.get(account.name):
+            raise DuplicateException(account, self)
+        self._accounts.append(account)
+
+    def get_account(self, name):
+        """
+        Retrieve an account from the state
+
+        :return: an account
+        :rtype: Account
+        """
+        return self.accounts.get(name)
+
+    def __repr__(self):
+        return f"State<{self.name}>"
+
+
+class Account(object):
+    """
+    Account represents an account and its DNS Zones
+
+    :param name: unique name that describes the account
+    :type name: str
+    """
+    def __init__(self, name):
+        self._name = name
+        self._zones = []
+
+    @property
+    def name(self):
+        """Returns the name of the account"""
+        return self._name
+
+    @property
+    def zones(self):
+        """
+        Returns the DNS zones managed under the account
+
+        :return: list of DNS zones
+        :rtype: list of Zone
+        """
+        return {a.name: a for a in self._zones}
+
+    def add_zone(self, zone):
+        """
+        Add a zone to the account
+
+        :param zone: DNS zone to add
+        :type zone: Zone
+        """
+        if self.zones.get(zone.name):
+            raise DuplicateException(zone, self)
+        self._zones.append(zone)
+
+    def get_zone(self, name):
+        """
+        Retrieve a zone by name
+
+        :param name: Zone name to retrieve
+        :type name: str
+        :return: DNS zone
+        :rtype: Zone
+        """
+        return self.zones.get(name)
+
+    def __repr__(self):
+        return f"Account<{self.name}>"
+
+
+class Zone(object):
+    """
+    Zone represents a DNS zone
+
+    :param name: DNS domain name
+    :param data: DNS zone data
+    :type name: str
+    :type data: dict
+    """
+    def __init__(self, name, data={}):
+        self._name = name.lower().rstrip('.')
+        self._records = []
+        self._data = data
+
+    @property
+    def data(self):
+        """Returns the zone data"""
+        return self._data
+
+    @property
+    def name(self):
+        """Returns the zone name"""
+        return self._name
+
+    @property
+    def records(self):
+        """Return DNS records"""
+        return {f"{r.name}": r for r in self._records}
+
+    def add_record(self, record):
+        """
+        Add a record to the zone
+
+        :param record: the record to add
+        :type record: Record
+        :raises DuplicateException: if a record by that name already exists
+        """
+        if self.records.get(record.name):
+            raise DuplicateException(record, self)
+        self._records.append(record)
+
+    def __eq__(self, other):
+        if not isinstance(other, Zone):
+            return False
+        return self.name == other.name
+
+    def __hash__(self):
+        return f"{self.name}".__hash__()
```

Comment:
This is defined because you want this object to be part of a `set()`. And you only need the `set()` because you wanted to use the `-` operand to find the diff between sets. If you do the refactor that I suggested in the caller, this `__hash__` method can be dropped.
Otherwise, if you keep the `__hash__`, by adjusting the `__eq__` implementation as I suggested above, the `__hash__` implementation should also be adjusted so it returns the same value to objects that evaluate to `True` when compared using `==`, meaning you have to also consider the records and the data to compute the hash.

Reply:
I've removed all `__hash__` methods in 8e3c392112e56cd517ab230cdd4314f532e3e5d3 as it's not using sets anymore